### PR TITLE
Show proper progress dialog when performing 'fix all in solution'

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/NoOpProgressTracker.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/NoOpProgressTracker.cs:5]: Can this be enabled?

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/SyntaxEditorBasedCodeFixProvider.FixAllProvider.cs [hunk 2]
```diff
@@ -33,26 +35,24 @@ public SyntaxEditorBasedFixAllProvider(SyntaxEditorBasedCodeFixProvider codeFixP
...
 
                 // Filter out any documents that we don't have any diagnostics for.
                 return result.Where(kvp => !kvp.Value.IsDefaultOrEmpty).ToImmutableDictionary();
-
-                static async Task<ImmutableDictionary<Document, ImmutableArray<Diagnostic>>> GetDocumentDiagnosticsToFixWorkerAsync(FixAllContext fixAllContext)
-                {
-                    return await FixAllContextHelper.GetDocumentDiagnosticsToFixAsync(
-                        fixAllContext,
-                        progressTrackerOpt: null).ConfigureAwait(false);
-                }
```
Comment (CyrusNajmabadi) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/SyntaxEditorBasedCodeFixProvider.FixAllProvider.cs:46]: i just inlined this method.  there was no point in this local function indirection.

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/NoOpProgressTracker.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Utilities/NoOpProgressTracker.cs:5]: yup yup.

File: src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs [hunk 3]
```diff
@@ -105,6 +120,21 @@ internal partial class BatchFixAllProvider : FixAllProvider
...
+            FixAllState fixAllState, IProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            try
+            {
+                await this.AddDocumentFixesAsync(document, diagnostics, fixes, fixAllState, cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                progressTracker.ItemCompleted();
+            }
```
Comment (davidwengier) [inline src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/BatchFixAllProvider.cs:135]: This pattern has been added a fair few times in this PR, might be worth an extension method to make it easier, so the caller of this method could just do something like:

```cs
using (var _ = progressTracker.ProcessItem())
{
	await this.AdddocumentFixesAsync(document, diagnostics, fixes, fixAllState, cancellationToken).ConfigureAwait(false);
}
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs [hunk 4]
```diff
@@ -48,24 +61,20 @@ internal static class FixAllContextHelper
...
-                        var projectToFix = projectsToFix[i];
-                        tasks[i] = Task.Run(async () =>
-                        {
-                            var projectDiagnostics = await fixAllContext.GetAllDiagnosticsAsync(projectToFix).ConfigureAwait(false);
-                            diagnostics.TryAdd(projectToFix.Id, projectDiagnostics);
-                            progressTrackerOpt?.ItemCompleted();
-                        }, cancellationToken);
-                    }
+                        foreach (var projectToFix in projectsToFix)
+                            tasks.Add(Task.Run(async () => await AddDocumentDiagnosticsAsync(diagnostics, projectToFix).ConfigureAwait(false), cancellationToken));
```
Comment (davidwengier) [inline src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/FixAllContextHelper.cs:73]: No need for this async state machine, right?

**Review by davidwengier — APPROVED** [review]

**Comment by sharwell** [comment]
@CyrusNajmabadi Can you update the pull request initial description to describe the behavior before and after this change?

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi Can you update the pull request initial description to describe the behavior before and after this change?

Sorry, didn't see this.  I will add.

**Pull request merged**